lpc55_isp/src/bin/lpc55_flash.rs

```diff
@@ -341,19 +350,46 @@ fn main() -> Result<()> {
             out.write_all(&m)?;
             eprintln!("CMPA Output written to {:?}", file);
         }
-        ISPCommand::ReadCFPA { file } => {
+        ISPCommand::ReadCFPA { page, file } => {
             do_ping(&mut *port)?;
 
-            let m = do_isp_read_memory(&mut *port, 0x9de00, 512)?;
+            let data = if let Some(page) = page {
+                // Only read one page as requested
+                let addr = match page {
+                    CfpaChoice::Scratch => 0x9de00,
+                    CfpaChoice::Ping => 0x9e000,
+                    CfpaChoice::Pong => 0x9e200,
+                };
+                do_isp_read_memory(&mut *port, addr, 512)?
+            } else {
+                // Read ping and pong pages and only write out the latest one.
+                let ping = do_isp_read_memory(&mut *port, 0x9e000, 512)
+                    .context("reading CFPA ping page")?;
+                let pong = do_isp_read_memory(&mut *port, 0x9e000, 512)
```

Comment:
Should the address here be 0x9e200? Looks to me like it's reading ping again.

Reply:
Ayup.  Probably should use constants for those base addresses.

Reply:
Thanks! Good catch.